src/server/scripts/Shadowlands/SanctumOfDomination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,443 @@
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 13, 484ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 13:
+                    me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SYLVANAS_INTRODUCTION_TALK_07, 0, 0);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 14, 5s + 516ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 14:
+                    me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SYLVANAS_INTRODUCTION_TALK_08, 0, 0);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 15, 1s + 516ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 15:
+                    me->NearTeleportTo(SylvanasIntroPos[3], false);
+                    me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SYLVANAS_TELEPORT, 0, 0);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 16, 1s + 265ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 16:
+                    if (Creature* bolvar = instance->GetCreature(DATA_BOLVAR_FORDRAGON_PINNACLE))
+                        me->SetFacingToObject(bolvar);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 17, 969ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 17:
+                    me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SYLVANAS_INTRODUCTION_TALK_09, 0, 0);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 18, 4s + 766ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 18:
+                    me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SYLVANAS_INTRODUCTION_TALK_10, 0, 0);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 19, 3s + 250ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 19:
+                    DoCastSelf(SPELL_RANGER_BOW_STANCE, false);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 20, 16ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 20:
+                    DoCastSelf(SPELL_GENERIC_ANCHOR_HERE, true);
+                    instance->SetData(DATA_SYLVANAS_INTRO, DONE);
+                    break;
+            }
+        }
+    }
+};
+
+// Serverside - Sylvanas Windrunner's Conversation Introduction
+struct at_sylvanas_windrunner_introduction : AreaTriggerAI
+{
+    at_sylvanas_windrunner_introduction(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger),
+        _instance(at->GetInstanceScript()) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (!_instance || !unit->IsPlayer() || (_instance->GetData(DATA_SYLVANAS_INTRO) == IN_PROGRESS || _instance->GetData(DATA_SYLVANAS_INTRO) == DONE))
```

Comment:
Should be faster if you are checking if `_instance->GetData(DATA_SYLVANAS_INTRO)` is != the required state since its only triggered once anyway right?

Reply:
I assume that would be better? In that case, I should be using `_instance->GetData(DATA_SYLVANAS_INTRO) == NOT_STARTED`?

Reply:
Feedback applied!